padle/padle.py

```diff
@@ -0,0 +1,801 @@
+        if len(todays_guesses) == 0:
+            return await send_cancellation_message(ctx, "You haven't guessed yet!")
+        padle_menu = PADleScrollMenu.menu()
+        cur_page = ceil(len(todays_guesses) / 5) - 1
+        page_guesses = await PADleScrollViewState.do_queries(dbcog,
+                                                             todays_guesses[((cur_page) * 5):((cur_page + 1) * 5)])
+        state = PADleScrollViewState(ctx.author.id, PADleScrollMenu.MENU_TYPE, monster=monster,
+                                     cur_day_page_guesses=page_guesses, current_day=cur_day,
+                                     current_page=cur_page, num_pages=cur_page + 1)
+        message = await padle_menu.create(ctx, state)
+        await self.config.user(ctx.author).edit_id.set(message.id)
+        await self.config.user(ctx.author).channel_id.set(ctx.channel.id)
+
+    @padle.command()
+    async def giveup(self, ctx):
+        """Give up on today's PADle"""
+        if await self.ensure_active_game(ctx):
+            return
+        confirmation = await get_user_confirmation(ctx, "Are you sure you would like to give up?", timeout=20)
+        dbcog = await self.get_dbcog()
+        if not confirmation:
+            if confirmation is None:
+                await send_cancellation_message(ctx, "Confirmation timeout.")
+            return
+        monster = dbcog.get_monster(int(await self.config.padle_today()))
+        await self.config.user(ctx.author).done.set(True)
+        m_embed = EmbedView(EmbedMain(title=MonsterHeader.menu_title(monster).to_markdown(),
+                                      description="PADle #{}".format(await self.config.num_days()),
+                                      url=MonsterLink.ilmina(monster), color=discord.Color.red()),
+                            embed_thumbnail=EmbedThumbnail(MonsterImage.icon(monster.monster_id))).to_embed()
+        await ctx.send(embed=m_embed)
+        await ctx.send(f"Use `{ctx.prefix}padle score` to share your score!")
+        async with self.config.user(ctx.author).score() as scores:
+            scores.append("\N{CROSS MARK}")
+        async with self.config.all_scores() as all_scores:
+            all_scores.append("X")
+        cur_day = await self.config.num_days()
+        todays_guesses = await self.config.user(ctx.author).todays_guesses()
+        async with self.config.user(ctx.author).all_guesses() as all_guesses:
+            all_guesses[str(cur_day)] = todays_guesses
+
+    @padle.command(aliases=["share"])
+    async def score(self, ctx):
+        """Share your PADle score"""
+        prefix = ctx.prefix
+        if not await self.config.user(ctx.author).done():
+            await ctx.send("You have not done today's PADle yet!")
+            return
+        score = await self.config.user(ctx.author).score()
+        value = "X" if "\N{CROSS MARK}" in score else len(await self.config.user(ctx.author).todays_guesses())
```

Comment:
This is fine if it has some meaning, but this is really mixing types.  You should probably keep this as an int xor a string.  Maybe -1 as a constant to mean this?